internal/server/storage/drivers/driver_linstor_utils.go

```diff
@@ -0,0 +1,215 @@
+	}
+
+	err = linstor.Client.ResourceGroups.Delete(context.TODO(), d.config[LinstorResourceGroupNameConfigKey])
+	if err != nil {
+		return fmt.Errorf("Could not delete Linstor resource group : %w", err)
+	}
+
+	return nil
+}
+
+// getResourceDefinition returns a Resource Definition instance for the given Resource name.
+func (d *linstor) getResourceDefinition(resourceDefinitionName string) (*linstorClient.ResourceDefinition, error) {
+	linstor, err := d.state.Linstor()
+	if err != nil {
+		return nil, fmt.Errorf("Could not load Linstor client: %w", err)
+	}
+
+	resourceDefinition, err := linstor.Client.ResourceDefinitions.Get(context.TODO(), resourceDefinitionName)
+	if errors.Is(err, linstorClient.NotFoundError) {
+		return nil, nil
+	} else if err != nil {
+		return nil, fmt.Errorf("Could not find the resource definition: %w", err)
+	}
+
+	return &resourceDefinition, nil
+}
+
+// getResourceDefinitionName returns the Linstor resource definition name for a given `vol`.
+func (d *linstor) getResourceDefinitionName(vol Volume) string {
+	// TODO: set an unique ID on the name so we can later rename the resource at the
+	// Incus level, since Linstor does not support renaming
+	volName := vol.Name()
+	volType := vol.contentType
+
+	return fmt.Sprintf("%s-%s_%s", d.config[LinstorResourceGroupNameConfigKey], volName, volType)
+}
+
+// getLinstorDevPath return the device path for a given `vol`.
+func (d *linstor) getLinstorDevPath(vol Volume) (string, error) {
+	linstor, err := d.state.Linstor()
+	if err != nil {
+		return "", fmt.Errorf("Could not load Linstor client: %w", err)
+	}
+
+	resourceDefinitionName := d.getResourceDefinitionName(vol)
+
+	// Fetching all the nodes that contains the volume definition
+	nodes, err := linstor.Client.Nodes.GetAll(context.TODO(), &linstorClient.ListOpts{
+		StoragePool: []string{d.config[LinstorResourceGroupNameConfigKey]},
+		Resource:    []string{resourceDefinitionName},
+	})
+	if err != nil {
+		return "", fmt.Errorf("Unable to get the nodes for the resource definition: %w", err)
+	}
+
+	// NOTE: Since we're mapping a unique Volume Definition to a Incus Volume, the Volume Number always be set to 0
+	volume, err := linstor.Client.Resources.GetVolume(context.TODO(), resourceDefinitionName, nodes[0].Name, 0)
+	if err != nil {
+		return "", fmt.Errorf("Unable to get Linstor volume: %w", err)
+	}
+
+	// TODO: maybe we should get the device path from /dev/drbd/by-res/<name>/0 like Cinder does
+	// https://github.com/LINBIT/openstack-cinder/blob/linstor/master/cinder/volume/drivers/linstordrv.py#L1065
+	return volume.DevicePath, nil
+}
+
+// makeVolumeAvailable makes a volume available on the current node.
+func (d *linstor) makeVolumeAvailable(vol Volume) error {
+	l := d.logger.AddContext(logger.Ctx{"volume": vol.Name()})
+	l.Debug("Making volume available on node", logger.Ctx{"volume": vol.Name()})
+	linstor, err := d.state.Linstor()
+	if err != nil {
+		return fmt.Errorf("Could not load Linstor client: %w", err)
```

Comment:
This adds a prefix that does not really add any information to an already prefixed error